pyanaconda/dbus/observer.py

```diff
@@ -93,6 +93,22 @@ def service_unavailable(self):
         """
         return self._service_unavailable
 
+    def connect(self):
+        """Connect to the service without waiting."""
+        bus_proxy = DBus.get_dbus_proxy()
+
+        if not bus_proxy.NameHasOwner(self.service_name):
+            raise DBusObserverError("Service {} is not available."
+                                    .format(self._service_name))
+
+        self.watch()
+        self._enable_service()
+
+    def disconnect(self):
+        """Disconnect the service without waiting."""
```

Comment:
I would change the `without waiting` -> `immediately`. The same in doc for `connect()`.

Also is here a disconnect method with waiting?